tests/playbooks/tests_route_device.yml

```diff
@@ -0,0 +1,195 @@
+              route"
+
+        - name: Get the IPv6 routes from the route table main
+          command: ip -6 route
+          register: route_table_main_ipv6
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv6
+            routes
+          assert:
+            that:
+              - route_table_main_ipv6.stdout is search("2001:db6::4 via
+                2001:db8::1 dev ethtest0\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv6
+              route"
+        - name: Get the interface1 MAC address
+          command: cat /sys/class/net/{{ interface1 }}/address
+          register: interface1_mac
+          ignore_errors: true  # noqa ignore-errors
+          changed_when: false
+        - name: Configure the IP addresses and the route with only the MAC
+            address specified
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface1 }}"
+                mac: "{{ interface1_mac.stdout }}"
+                type: ethernet
+                autoconnect: false
+                ip:
+                  address:
+                    - 198.51.100.4/24
+                    - 2001:db8::6/32
+                  route:
+                    - network: 198.58.10.64
+                      prefix: 26
+                      gateway: 198.51.100.102
+                      metric: 4
+
+        - name: Assert that the warning about specifying the route without
+            the output device is logged for initscripts provider
+          assert:
+            that:
+              - __network_connections_result.stderr is search("The connection
+                .* does not specify an interface name. Therefore, the route .*
+                will be configured without the output device and the kernel
+                will choose it automatically which might result in an unwanted
+                device being used. To avoid this, specify `interface_name` in
+                the connection appropriately.")
+            msg: The warning about specifying the route without the output
+              device is not logged for initscripts provider
+          when: network_provider == "initscripts"
+
+        - name: Assert that the warning about specifying the route without
+            the output device is not logged for nm provider
+          assert:
+            that:
+              - __network_connections_result.stderr is not search("The
+                connection .* does not specify an interface name. Therefore,
+                the route .* will be configured without the output device and
+                the kernel will choose it automatically which might result in
+                an unwanted device being used. To avoid this, specify
+                `interface_name` in the connection appropriately.")
```

Comment:
can you assert that there is no warning in this case? Are there any warnings usually for this configuration?

Reply:
We can assert that there is no warning in this case. There is no warning for this configuration.